hack/update-vendor.sh

```diff
@@ -216,10 +242,6 @@ kube::util::list_staging_repos \
 
 # pin referenced versions
 ensure_require_replace_directives_for_all_dependencies
-# resolves/expands references in the root go.mod (if needed)
-go mod tidy >>"${LOG_FILE}" 2>&1
-# pin expanded versions
-ensure_require_replace_directives_for_all_dependencies
 # group require/replace directives
 group_directives
```

Comment:
we need to tidy before calling group_directives so any expanded/added require directives get grouped appropriately

the later tidy calls are only expected to *remove* things, but the first time we call it, we might be *adding* indirect deps